### PR TITLE
Add Missing config `enableOutOfMemoryTracking` for iOS/Mac

### PR DESCRIPTION
**Pull request by lucas-zimerman**
Add an option filter to avoid registering out of error events if the user doesn't want it.

Partially solves #127

### CODE REVIEW

File: src/options.ts [hunk 1]
```diff
@@ -35,4 +35,14 @@ export interface CapacitorOptions
...
   attachThreads?: boolean;
+
+  /**
+  * Enables Out of Memory Tracking for iOS and macCatalyst.
+  * See the following link for more information and possible restrictions:
+  * https://docs.sentry.io/platforms/apple/guides/ios/configuration/out-of-memory/
+  *
+  * @default true
+  * */
+  enableOutOfMemoryTracking?: boolean;
```
Comment (bruno-garcia) [inline src/options.ts:46]: Do we have this on the React Native SDK? I wonder if this is a sign we should have this TypeScript type definition live elsewhere and use it in both capacitor+RN
cc @marandaneto @jennmueng

Comment (lucas-zimerman) [reply src/options.ts:46]: Yes, it's also on React Native
https://github.com/getsentry/sentry-react-native/blob/3f0c1707d86c1b6e28582d1db21926f2ac543f0b/src/js/options.ts#L76-L83

Comment (marandaneto) [reply src/options.ts:46]: a Unified package would not be bad but each platform has its specific options, not sure how much we'd gain with it.
Not a bad idea tho.

**Comment by piotr-cz** [comment]
This option is missing in [Capacitor docs](https://docs.sentry.io/platforms/javascript/guides/capacitor/configuration/options/#hybrid-sdk-options) (available in [React Native docs](https://docs.sentry.io/platforms/react-native/configuration/options/#enableOutOfMemoryTracking))

**Comment by lucas-zimerman** [comment]
Thanks for pointing about the Documentation.
You can track the changes here: https://github.com/getsentry/sentry-docs/pull/5118

**Pull request merged**